auth/src/main/java/com/firebase/ui/auth/ui/phone/SpacedEditText.java

```diff
@@ -65,6 +65,11 @@ public void setText(CharSequence text, BufferType type) {
      */
     @Override
     public void setSelection(int index) {
+        // Prevent index from ever exceeding the bounds of the text (See #779)
+        if (index >= mOriginalText.length()) {
+            index = mOriginalText.length() - 1;
+        }
+
         // If the index is the leading edge, there are no spaces before it.
         // For all other cases, the index is preceded by index - 1 spaces.
         int spacesUpToIndex;
```

Comment:
Now thinking about this, we have:
``` 
index --> spacesUpToIndex:
0 --> 0
1 --> 0
2 --> 1
3 --> 2
4 --> 3
5 --> 4
```

But I actually think it should be:
``` 
index --> spacesUpToIndex:
0 --> 0
1 --> 1
2 --> 2
3 --> 3
4 --> 4
5 --> 5
```
Because index 5 is actually the 6th dash.

In the case where index == 5 (the max case) then this would call setSelection(10) which would select the final character of the 11-character string (zero-indexed)

Reply:
Thanks for that!  Implemented your suggestion but with some more comments and spacing for clarity.